post.go

```diff
@@ -19,6 +19,7 @@ func (p *PostService) CreatePost(post *model.Post) error {
 		return normalizeAppErr(appErr)
 	}
 
+	//nolint:govet //lock value is not relevant to consumer
```

Comment:
@lieut-data I wanted to have your attention on these two nolints. They are not introduced by my code, and I wanted to double check that the reasoning is in place.
The problem that the linter drops is:
> copylocks: assignment copies lock value to *post: github.com/mattermost/mattermost-server/v5/model.Post contains sync.RWMutex

Reply:
I think this was from an upstream change in mattermost-server -- I think we can handle this by using the new `Clone()` method on `model.Post`.